libs/withdraws/src/lib/use-ready-to-complete-withdrawals-toast.tsx

```diff
@@ -276,11 +278,29 @@ const SingleReadyToWithdrawToastContent = ({
       <Button
         data-testid="toast-complete-withdrawal"
         size="xs"
-        onClick={() => {
-          createEthWithdrawalApproval(
-            withdrawal,
-            approval?.erc20WithdrawalApproval
-          );
+        onClick={async () => {
+          const asset = withdrawal.asset;
+          if (
+            asset.source.__typename === 'ERC20' &&
+            asset.source.chainId !== String(chainId)
+          ) {
+            await connector.provider?.request({
+              method: 'wallet_switchEthereumChain',
+              params: [
+                {
+                  chainId: `0x${Number(asset.source.chainId).toString(16)}`,
+                },
+              ],
+            });
+          }
+
+          // without this timeout the the complete tx does not trigger, dont know why
+          setTimeout(() => {
+            createEthWithdrawalApproval(
+              withdrawal,
+              approval?.erc20WithdrawalApproval
+            );
+          }, 500);
```

Comment:
**nitpick (non-blocking)**: This is a pretty big, unwieldy `onClick` that would be better outside the JSX. But it's not worth blocking.

Reply:
Agree, but we'll have to come back to this.